test-harness/index.js

```diff
@@ -0,0 +1,270 @@
+const fetch = require('node-fetch');
+const requests = require('./requests');
+
+const { exec } = require('child_process');
+const { makeRequest, constructMasterFileName, readFile } = require('./helpers');
+
+async function waitForPrism(done) {
+  try {
+    await fetch('http://localhost:4010');
+    done();
+  } catch (err) {
+    setTimeout(() => {
+      waitForPrism(done);
+    }, 500);
+  }
+}
+
+async function runTest(req) {
+  const { dynamic, ...request } = req;
+
+  const masterFileName = constructMasterFileName(request);
+  const masterFile = readFile(masterFileName);
+
+  const reqRes = await makeRequest(request);
+
+  return {
+    reqRes,
+    masterFile,
+  };
+}
+
+function killPrism(done = () => null) {
```

Comment:
Hmm, I'm generally not a great fan of spawning processes from an application; I can live with it since it's testing code, but if there's another way, it'd be way better.

Reply:
Hey @XVincentX, so I was also trying to spawn `prism` with `circleci` step, but was having: https://github.com/circleci/circleci-docs/issues/1037
for some, it was "resolved" with adding a delay of 10 seconds or something, but I didn't find this to always be so